packages/core/botpress/src/web/components/Util/BackendToast.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import { toast } from 'react-toastify'
+import _ from 'lodash'
+
+import EventBus from '~/util/EventBus'
+
+export default class BackendToast extends Component {
+  constructor(props) {
+    super(props)
+
+    const uniqueCodes = _.uniq(props.codes)
+    uniqueCodes.forEach(code => {
+      let id = null
+
+      EventBus.default.on(`toast.${code}`, options => {
+        options.text = options.text.length > 64 ? `${options.text.slice(0, 63)}...` : options.text
+
+        id = BackendToast[options.type](options, id)
+      })
+    })
+  }
+
+  static dismissIfActive = id => toast.isActive(id) && toast.dismiss(id)
+
+  static info = ({ text, time }, id) => {
+    BackendToast.dismissIfActive(id)
+
+    return toast.info(text, { autoClose: time })
+  }
+
+  static success = ({ text }, id) => {
+    BackendToast.dismissIfActive(id)
+
+    return toast.success(text)
+  }
+
+  static error = ({ text }, id) => {
```

Comment:
If I understand it correctly this means that whenever the new error happens it will dismiss any other existing toast. Not sure if it's the real problem but logically I have some doubts it's what we want (similarly for all other toast types).

As I said I can be missing the idea so if that's really logical please explain.

Nevertheless, I'm approving the PR now as the rest looks good to me.

Reply:
@emirotin no at all. If user create few QnA and some of this QnA have error, it will replace the toast. But if you process NLU sync and create QnA it will show two error.